panther_analysis_tool/rule.py

```diff
@@ -184,25 +252,32 @@ def __init__(self, config: Mapping):
                 analysisType: either RULE or SCHEDULED_RULE
                 id: Unique rule identifier
                 body: The rule body
-                (Optional) version: The version of the rule
+                versionId: The version of the rule
+                (Optional) path: The rule module path
                 (Optional) dedupPeriodMinutes: The period during which
-                    the events will be deduplicated
+                the events will be deduplicated
         """
         self.logger = get_logger()
 
         # Check for required string fields
-        for each_field in ["id", "body", "versionId"]:
+        for each_field in ["id", "versionId"]:
             if not (each_field in config) or not isinstance(config[each_field], str):
                 raise AssertionError('Field "%s" of type str is required field' % each_field)
 
+        if not (config.get("body") or config.get("path")):
+            raise ValueError('one of "body", "path" must be defined')
```

Comment:
I am not sure if type checks are a usual practice in Python in general, that's why I omitted it. But I'll add it back as you mention, no harm in getting a more precise error since these are critical parameters.

Reply:
Sounds good. And really I'm open to having it in both places or not having it in both places, with type hints it doesn't seem so critical I just want us to be consistent.

Reply:
Done in 53eff3c92d3943935a54a95a650d352c9b6da1c7, with tests.